go/cmd/dolt/commands/blame.go

```diff
@@ -84,11 +94,44 @@ func (cmd BlameCmd) Exec(ctx context.Context, commandStr string, args []string,
 	help, usage := cli.HelpAndUsagePrinters(cli.CommandDocsForCommandString(commandStr, blameDocs, ap))
 	apr := cli.ParseArgsOrDie(ap, args, help)
 
-	if apr.NArg() != 1 {
+	if apr.NArg() > 2 || apr.NArg() == 0 {
 		usage()
 		return 1
 	}
-	args = []string{"--" + QueryFlag, fmt.Sprintf(blameQueryTemplate, apr.Arg(0))}
 
-	return SqlCmd{}.Exec(ctx, "sql", args, dEnv, cliCtx)
+	queryist, sqlCtx, closeFunc, err := cliCtx.QueryEngine(ctx)
+	if err != nil {
+		iohelp.WriteLine(cli.CliOut, err.Error())
+		return 1
+	}
+	if closeFunc != nil {
+		defer closeFunc()
+	}
+
+	var schema sql.Schema
+	var ri sql.RowIter
+	if apr.NArg() == 1 {
+		schema, ri, err = queryist.Query(sqlCtx, fmt.Sprintf(blameQueryTemplate, apr.Arg(0), "HEAD"))
+	} else {
+		// validate input
+		_, err = ResolveCommitWithVErr(dEnv, apr.Arg(0))
```

Comment:
See chat discussion. This is a very reliable way to validate the input - but it used dEnv which is off limits.